Source/global.js

```diff
@@ -174,6 +176,80 @@ class global {
     get usagePrefix() {
         return '\n\t ';
     }
+    /**
+     * Gets the namespace based on the closest csprojPath and the cwd path
+     * @param {String} currentPath 
+     * @param {String} csprojPath 
+     * @returns {String}
+     */
+    createCSharpNamespace(currentPath, csprojPath) {
+        const csprojFileName = path.parse(this.getFileName(csprojPath)).name;
+        const csprojFileDir = this.getFileName(this.getFileDir(csprojPath));
+        let namespaceSegments = [];
+        
+        let segmentPath = currentPath;
+        let segment = this.getFileName(segmentPath);
+
+        console.log('csprojFileName: ', csprojFileName);
+        console.log('csprojFileDir: ', csprojFileDir);
+        while (segment != csprojFileDir) {
+
+            console.log('segmentPath: ', segmentPath);
+            console.log('segment: ', segment);
+            namespaceSegments.push(segment);
+            segmentPath = this.getFileDir(segmentPath);
+            segment = this.getFileName(segmentPath);
+        } 
+        namespaceSegments = namespaceSegments.reverse();
+        
+        let namespace = csprojFileName;
+        namespaceSegments.forEach(element => {
+            namespace += '.' + element;
+        });
+        return namespace;
+    }
+    /**
+     * Gets the path of the nearest .csproj file, searching upwards not recursively
+     */
+    getNearestCsprojFile() {
+        let currentPath = process.cwd();
```

Comment:
I know this is changed in a later commit - but want to point out; using `process` directly like this makes it virtually impossible to consistently create tests for it. Which leads me to; there are to specs for this :)